compiler/src/dmd/dsymbolsem.d

```diff
@@ -2698,6 +2698,10 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
                 em.origValue = e;
             }
             em.value = e;
+            // https://issues.dlang.org/show_bug.cgi?id=24311
+            // First enum member is .init value, which gets put into static segment
+            if (first)
+                lowerStaticAAs(e, sc);
```

Comment:
Maybe this branch could be taken only if the enum member type is an associative array? That way we avoid the cost of traversing the AST for enums that do not contain static arrays.

Reply:
Checking that the type is an associative array is not sufficient, it could also be a static array of structs with a field that's a pointer to an associative array. Doing a comprehensive type check is complex and takes a similar amount of time to `lowerStaticAAs`. 

Named enums are usually simple integer enumerations without an explicit first member value, and if they do have one, it's usually just an integer or identifier, which doesn't take long to walk.

Reply:
> Named enums are usually simple integer enumerations without an explicit first member value

Yeah, that's what I was thinking. To me it seems that allowing enum members to have such complicated forms ends up slowing compilation due to all of the extra cases, while in the majority of cases, folks just use integer enums.

However, at this point, indeed, this seems like the correct fix.